source/parser.hera

```diff
@@ -1594,6 +1595,80 @@ MethodDefinition
       parameters: signature.parameters,
     }
 
+Constructor
+  # NOTE: Not adding extra validation using PropertySetParameterList
+  # NOTE: If this node layout changes, be sure to update `convertMethodTOFunction`
+  Decorators?:decorators ConstructorSignature:signature BracedBlock:block ->
+    return {
+      type: "Constructor",
+      children: $0,
+      name: signature.name,
+      decorators,
+      signature,
+      block,
+      parameters: signature.parameters,
+    }
+
+ConstructorSignature
+  # NOTE: If this node layout changes, be sure to update `convertMethodTOFunction`
+  MethodModifier? "constructor" _* ConstructorParameters:parameters ReturnTypeSuffix?:suffix ->
```

Comment:
`ConstructorSignature` can't have `MethodModifier` or `ReturnTypeSuffix` so those can be removed here.

Also there is a `@` constructor shorthand that should be added as an alternative to `"constructor"`.